web/components/answers/answer-resolve-panel.tsx

```diff
@@ -419,12 +422,49 @@ export const IndependentAnswersResolvePanel = (props: {
 
   const isAdmin = useAdmin()
   const user = useUser()
+  const answers = useAnswersCpmm(contract.id) ?? contract.answers
+  const isAdminorMod = useAdminOrMod()
+
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [error, setError] = useState<string | undefined>(undefined)
+
+  const onCancelAll = async () => {
+    setIsSubmitting(true)
+    try {
+      for (const answer of answers) {
+        const resolutionProps = removeUndefinedProps({
+          contractId: contract.id,
+          outcome: 'CANCEL',
+          answerId: answer.id,
+        })
+
+        try {
+          const result = await api(
+            'market/:contractId/resolve',
+            resolutionProps as any
+          )
+          console.log('resolved', resolutionProps, 'result:', result)
+        } catch (e) {
+          if (e instanceof APIError) {
+            setError(e.toString())
+          } else {
+            console.error(e)
+            setError('Error resolving question')
+          }
+        }
+      }
+      console.log('All contracts canceled')
+    } catch (e) {
+      console.error(e)
+      setError('Error canceling all contracts')
+    }
+    setIsSubmitting(false)
+  }
 
-  const { answers, addAnswersMode } = contract
   const sortedAnswers = sortBy(
     answers,
     (a) => (a.resolution ? -a.subsidyPool : -Infinity),
-    (a) => (addAnswersMode === 'ANYONE' ? -1 * a.prob : a.index)
+    (a) => a.index
```

Comment:
Are you sure we want this? Actually, we should try to use the same sort as the answers panel.

Use the `sortAnswers` function from common/answer instead of all this.